mozdownload/scraper.py

```diff
@@ -776,6 +782,10 @@ def __init__(self, branch='mozilla-central', build_number=None, date=None,
     def get_build_info(self):
         "Defines additional build information"
 
+        # Find branch by application
+        if self.branch is None:
+            self.branch = APPLICATIONS[self.application]
```

Comment:
Oh, please drop my comment from above. The base Scraper class doesn't need the branch property. So what you need to do here is to have this code in the `__init__` method of `DailyScaper` but after line 384 which initializes the `Scraper` class first.